### PR TITLE
Take into account application context path and metrics path

### PR DESCRIPTION
**Pull request by jmartisk**
https://github.com/quarkusio/quarkus/issues/3745

### CODE REVIEW

File: extensions/smallrye-metrics/runtime/src/main/java/io/quarkus/smallrye/metrics/runtime/SmallRyeMetricsServlet.java [hunk 1]
```diff
@@ -18,18 +20,23 @@
...
+    String appContextPath;
+
     @Override
     protected void doOptions(HttpServletRequest req, HttpServletResponse resp) throws IOException {
         doGet(req, resp);
     }
 
     @Override
     protected void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String metricsPath = getInitParameter("metrics.path");
```
[the remark this answers is not in the document]
Comment (jmartisk) [reply extensions/smallrye-metrics/runtime/src/main/java/io/quarkus/smallrye/metrics/runtime/SmallRyeMetricsServlet.java]: This is an immutable property that is passed to the servlet at build time, so it should not be necessary to re-compute anything

[the remark this answers is not in the document]
Comment (jmartisk) [reply extensions/smallrye-metrics/runtime/src/main/java/io/quarkus/smallrye/metrics/runtime/SmallRyeMetricsServlet.java]: Oh sorry, now I understand your point, yeah I will move this to `init`

File: extensions/smallrye-metrics/runtime/src/main/java/io/quarkus/smallrye/metrics/runtime/SmallRyeMetricsServlet.java [hunk 2]
```diff
@@ -18,18 +20,23 @@
...
+        String contextPath = (appContextPath.equals("/") ? "" : appContextPath) + metricsPath;
```
Comment (gsmet) [inline extensions/smallrye-metrics/runtime/src/main/java/io/quarkus/smallrye/metrics/runtime/SmallRyeMetricsServlet.java]: I think we need to be a bit more careful about the `appContextPath` ending with a `/` or not?

Comment (jmartisk) [reply extensions/smallrye-metrics/runtime/src/main/java/io/quarkus/smallrye/metrics/runtime/SmallRyeMetricsServlet.java]: Ok I guess just stripping off a trailing `/` from `appContextPath` will  do the work

File: extensions/smallrye-metrics/runtime/src/main/java/io/quarkus/smallrye/metrics/runtime/SmallRyeMetricsServlet.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (jmartisk) [reply extensions/smallrye-metrics/runtime/src/main/java/io/quarkus/smallrye/metrics/runtime/SmallRyeMetricsServlet.java]: Done

File: extensions/smallrye-metrics/runtime/src/main/java/io/quarkus/smallrye/metrics/runtime/SmallRyeMetricsServlet.java (hunk 2 above)
Comment (jmartisk) [reply extensions/smallrye-metrics/runtime/src/main/java/io/quarkus/smallrye/metrics/runtime/SmallRyeMetricsServlet.java]: Done

File: extensions/smallrye-metrics/runtime/src/main/java/io/quarkus/smallrye/metrics/runtime/SmallRyeMetricsServlet.java [hunk 3]
```diff
@@ -18,6 +20,21 @@
...
+    // full path where the metrics endpoint resides, by default it's /metrics, but can be /appContextPath/anything
+    private String contextPath;
+
+    @Override
+    public void init() {
+        String metricsPath = getInitParameter("metrics.path");
+        // strip off trailing / if there is
+        String appContextPathStripped = appContextPath.endsWith("/") ? appContextPath.substring(0, appContextPath.length() - 1)
+                : appContextPath;
+        contextPath = appContextPathStripped + metricsPath;
```
[the remark this answers is not in the document]
Comment (jmartisk) [reply extensions/smallrye-metrics/runtime/src/main/java/io/quarkus/smallrye/metrics/runtime/SmallRyeMetricsServlet.java]: Ok I added some additional sanitizing so now it handles 
- the app context path with or without a trailing `/`
- the metrics endpoint with or without a leading `/`

**Review by kenfinnigan — APPROVED** [review]

**Pull request merged**